src/cache.rs

```diff
@@ -86,97 +102,335 @@ impl LeaseCache {
+            t1: 0,
+            t2: 3600,
+            lease_time: 0,
+            mtu: 0,
+            domain_name: "example.domain".to_string(),
+            mac_address: String::from(mac_address),
+            siaddr: random_ipv4(),
+            yiaddr: random_ipv4(),
+            srv_id: random_ipv4(),
+            subnet_mask: "".to_string(),
+            broadcast_addr: "".to_string(),
+            dns_servers: vec![],
+            gateways: vec![],
+            ntp_servers: vec![],
+            host_name: "example.host_name".to_string(),
+            is_v6: false,
+        }
     }
 
     #[test]
-    fn update() {
-        // 1. Clean the directory to the lease
-        // 2. Create a new cache instance
-        // 3. Check that the file to the cache exists
-        // 4. Change the value of that lease and call the update method
-        // 5. Check that the old lease doesnt exist and the new lease is up to date
-        // 6. Clean the directory to the lease
+    fn add_leases() {
+        // Instead of using a file use a cursor
+        let buff = Cursor::new(Vec::new());
+        let mut cache = match LeaseCache::new(buff) {
+            Ok(cache) => cache,
+            Err(e) => panic!("Could not create leases cache: {:?}", e),
+        };
+
+        // Create a random amount of randomized leases
+        let mut macaddrs = Vec::new();
+        let mut rng = thread_rng();
+        // Make a random amount of leases
+        let range: u8 = rng.gen_range(0..10);
+        for i in 0..range {
+            // Create a random mac address to create a random lease of that mac address
+            let mac_address = random_macaddr().to_string();
+            macaddrs.push(mac_address.clone());
+            let lease = random_lease(&mac_address);
+
+            // Add the lease to the cache
+            cache
+                .add_lease(&mac_address, &lease)
+                .expect("could not add lease to cache");
+
+            // Deserialize the written bytes to compare
+            let lease_bytes = cache.writer.get_ref().as_slice();
+            let s: HashMap<String, Vec<NetavarkLease>> = match serde_json::from_slice(lease_bytes) {
+                Ok(s) => s,
+                Err(e) => panic!("Error: {:?}", e),
+            };
+
+            // Get the mac address of the lease
+            let macaddr = macaddrs
+                .get(i as usize)
+                .expect("Could not get the mac address of the lease added");
+
+            // Find the lease in the set of deserialized leases
+            let deserialized_lease = s
+                .get(macaddr)
+                .expect("Could not get the mac address from the map")
+                .get(0)
+                .expect("Could not get lease from set of mac addresses")
+                .clone();
+            // Assure that the amount of leases added is correct amount
+            assert_eq!(s.len(), (i + 1) as usize);
+            // Assure that the lease added was correct
```

Comment:
i'd like to see you add a test for two identical leases

Reply:
Can you explain what you mean by this comment?

Reply:
i dont see the issue you were supposed to record.